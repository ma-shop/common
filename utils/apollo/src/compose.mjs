export { flowRight as compose } from 'lodash'
