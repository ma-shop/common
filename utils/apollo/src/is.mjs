import { is } from '@ma-shop/is'

/// @name network
/// @author Alex Andreescu
/// @description
/// contains the common networkStatus checks that are done throughout the app
/// @example
/// const networkStatus = 3
/// return is.network.fetching(networkStatus) // returns true
/// @note
/// the following documentation for network is summarized from the github link below
/// https://github.com/apollographql/apollo-client/blob/master/packages/apollo-client/src/core/networkStatus.ts
is.network = {
  ///# @name loading
  ///# @author Alex Andreescu
  ///# @description
  ///# determines if a query was fired if not fired before
  ///# @arg {int} networkStatus
  ///# @returns {boolean} eg. true | false
  ///# @note this will be the status if a query fires with partial data returned from the cache
  loading (networkStatus = 0) {
    return networkStatus === 1
  },
  ///# @name fetching
  ///# @author Glyn Thomas
  ///# @description
  ///# Takes a networkStatus integer and returns a boolean
  ///# identifiying if that networkStatus is fetching
  ///# @arg {int} networkStatus
  ///# @returns {boolean} eg. true | false
  fetching (networkStatus = 0) {
    return networkStatus === 3
  },
  ///# @name refetching
  ///# @author Alex Andreescu
  ///# @description
  ///# determines if a query is being refetched
  ///# @note includes generic request flags
  ///# @arg {int} networkStatus
  ///# @returns {boolean} eg. true | false
  refetching (networkStatus = 0) {
    return [ 1, 2, 4 ].includes(networkStatus)
  },
  ///# @name finished
  ///# @author Alex Andreescu
  ///# @description
  ///# determines if a query is no longer inflight
  ///# @note agnostic of response status
  ///# @arg {int} networkStatus
  ///# @returns {boolean} eg. true | false
  finished (networkStatus = 0) {
    return networkStatus > 6
  },
  ///# @name success
  ///# @author Alex Andreescu
  ///# @description
  ///# determines if a query was successfully finished
  ///# @arg {int} networkStatus
  ///# @returns {boolean} eg. true | false
  success (networkStatus = 0) {
    return networkStatus === 7
  },
  ///# @name error
  ///# @author Alex Andreescu
  ///# @description
  ///# determines if a query threw an error
  ///# @arg {int} networkStatus
  ///# @returns {boolean} eg. true | false
  error (networkStatus = 0) {
    return networkStatus === 8
  },
}
