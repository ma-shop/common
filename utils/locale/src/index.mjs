// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// !!! This file is generated do not modify it manually because it will be overwritten
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export * from './locale'
export * from './locales'
