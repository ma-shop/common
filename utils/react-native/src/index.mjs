// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// !!! This file is generated do not modify it manually because it will be overwritten
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export * from './get-device-locale'
export * from './interactions'
export * from './is'
export * from './navigation'
export * from './shapes'
