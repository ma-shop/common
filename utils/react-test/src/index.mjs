// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// !!! This file is generated do not modify it manually because it will be overwritten
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export * from './enzyme'
export * from './react-test'
export * from './render-items'
