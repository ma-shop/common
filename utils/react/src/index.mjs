// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// !!! This file is generated do not modify it manually because it will be overwritten
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export * from './hoist-static'
export * from './map-props'
export * from './render-if'
export * from './set-ref'
export * from './set-state-async'
export * from './shapes'
export * from './use-previous'
