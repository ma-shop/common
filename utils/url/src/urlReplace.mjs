/// @name urlReplace
/// @author Cameron Bates
/// @description
/// Uses an object of key/values to replace templated values out of
/// a url string
/// @param {string} url - string with templates to be replaced
/// @param {object} values - object of key/value pairs
export function urlReplace (url = '', values = {}) {
  const keysArray = Object.keys(values)
  const regex = new RegExp(`{{(${keysArray.join('|')})}}`, 'g')

  const urlPlaceholders = url.match(/{{.*?}}/g)
  if (urlPlaceholders.length !== keysArray.length) {
    throw new Error(`Length mismatch, keys potentially incorrect.\n
    Url Placeholders(${urlPlaceholders.length}): ${urlPlaceholders.join()}\n
    Values(${keysArray.length}): ${keysArray.join()}`)
  }

  return url.replace(regex, (_, key) => values[key])
}
