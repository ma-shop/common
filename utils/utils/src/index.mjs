// @note this is not automated

export * from './delay'
export * from './lodash'
export * from './noop'
export * from './normalize'
export * from './round'
export * from './set-env'
export * from './ui'

// @note this has to be last because jest is weird
export * from './packages'
