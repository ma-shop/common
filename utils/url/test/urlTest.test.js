import { urlReplace } from '../src'


describe('urlReplace', () => {
  test('takes in an object with one value and returns the correct string', () => {
    const originalUrl = 'Site/Search/Suggest/{{query}}'
    const replacedUrl = 'Site/Search/Suggest/isotonix'
    const keys = { query: 'isotonix' }

    expect(urlReplace(originalUrl, keys)).toEqual(replacedUrl)
  })

  // eslint-disable-next-line max-len
  test('takes in an object with multiple keys and returns the correct string even if the keys are out of order', () => {
    const originalUrl = [
      'Store/Listings',
      '/StoreType/{{storeType}}',
      '/StoreCategories/{{storeCategories}}',
      '/Letter/{{letter}}',
      '/Term/{{term}}',
      '/Page/{{page}}',
      '/{{sort}}',
      '/City/{{city}}',
    ].join('')
    const replacedUrl = 'Store/Listings/StoreType/OneCart/StoreCategories/Auto/Letter/C/Term/tires/Page/1/NAME-ASC/City/Canada'
    const keys = {
      sort: 'NAME-ASC',
      page: '1',
      storeCategories: 'Auto',
      term: 'tires',
      storeType: 'OneCart',
      city: 'Canada',
      letter: 'C',
    }

    expect(urlReplace(originalUrl, keys)).toEqual(replacedUrl)
  })

  // eslint-disable-next-line max-len
  test('takes in an object with more than one key but fails because the string only has one value to replace', () => {
    const originalUrl = 'Store/Product/{{prodId}}'
    const keys = {
      prodId: '12345',
      allPerms: true,
    }

    expect(() => urlReplace(originalUrl, keys)).toThrowErrorMatchingSnapshot()
  })

  // eslint-disable-next-line max-len
  test('takes in an object with one key but fails because the string only has two values to replace', () => {
    const originalUrl = 'Store/Product/{{prodId}}/{{allPerms}}'
    const keys = {
      prodId: '12345',
    }

    expect(() => urlReplace(originalUrl, keys)).toThrowErrorMatchingSnapshot()
  })
})
