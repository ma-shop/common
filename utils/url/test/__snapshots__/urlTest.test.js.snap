// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`urlReplace takes in an object with more than one key but fails because the string only has one value to replace 1`] = `
"Length mismatch, keys potentially incorrect.

    Url Placeholders(1): {{prodId}}

    Values(2): prodId,allPerms"
`;

exports[`urlReplace takes in an object with one key but fails because the string only has two values to replace 1`] = `
"Length mismatch, keys potentially incorrect.

    Url Placeholders(2): {{prodId}},{{allPerms}}

    Values(1): prodId"
`;
